src/views/MarketDetails/CurrentMarketDetails2.tsx

```diff
@@ -0,0 +1,154 @@
+import { selectCurrentMarketInfo } from '@/abacus-ts/selectors/markets';
+import { IndexerPerpetualMarketType } from '@/types/indexer/indexerApiGen';
+import BigNumber from 'bignumber.js';
+import { shallowEqual } from 'react-redux';
+
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useMetadataServiceAssetFromId } from '@/hooks/useMetadataService';
+import { useStringGetter } from '@/hooks/useStringGetter';
+
+import { DetailsItem } from '@/components/Details';
+import { DiffOutput } from '@/components/DiffOutput';
+import { Output, OutputType } from '@/components/Output';
+
+import { useAppSelector } from '@/state/appTypes';
+
+import { BIG_NUMBERS } from '@/lib/numbers';
+import { orEmptyObj } from '@/lib/typeUtils';
+
+import { MarketDetails } from './MarketDetails';
+
+export const CurrentMarketDetails = () => {
+  const stringGetter = useStringGetter();
+  const currentMarketData = useAppSelector(selectCurrentMarketInfo, shallowEqual);
+  const asset = useMetadataServiceAssetFromId(currentMarketData?.ticker);
```

Comment:
why not use the asset data in the state? Is it missing stuff?

Reply:
Maybe we should replace the assets in raw state with the query inside useMetadataService actually.